inloop/solutions/views.py

```diff
@@ -192,7 +192,12 @@ class SolutionArchiveDownloadView(LoginRequiredMixin, View):
     def get(self, request, solution_id):
         if not solution_id:
             raise Http404('No solution id was supplied.')
-        solution = get_object_or_404(Solution, pk=solution_id, author=request.user)
+        solution = get_object_or_404(Solution, pk=solution_id)
+        if request.user.id != solution.author_id and not request.user.is_staff:
+            raise Http404(
+                'Only the author himself and staff members '
+                'are allowed to download this archive.'
+            )
```

Comment:
```suggestion
        if not request.user.is_staff:
            solution = get_object_or_404(Solution, pk=solution_id, author=request.user)
        else:
            solution = get_object_or_404(Solution, pk=solution_id)
```
… and skip the error message alltogether, it leaks too much information and `404 Not Found` is enough.